### PR TITLE
Display graphical statistics of delegates - closes #2623

### LINKED ISSUE #2623: Display graphical statistics of delegates
### Description
As a user; I'd like to see a graphical overview of the delegate activities on the delegates page.

The graphs include:
1. Percentage/number of active delegates vs standby delegates. 
2. Delegates forging status.
3. Registered delegates.

### Acceptance criteria
- the values must not change when user switches the tab in the list of delegates from active to standby.
- The graphs must start with a slight animation.
- The data of forging status is related to active delegates.
- The registered delegates graph shows only the last 4 month or the last 100 transaction. which ever comes earlier.
- Doughnut charts have a tooltip.

### Design
https://projects.invisionapp.com/d/main#/console/18539694/389182920/preview

### PR DESCRIPTION
**Pull request by osvaldovega**
### What issue have I solved?

-- #2623 

### How have I implemented/fixed it?

- [x] Create Overview component for delegates tab
- [x] Add logic for parse data for each chart, as the data need to be normalize.
- [x] Add 2 more API calls for the charts (Active and Standby delegates and registered delegates)
- [x] Update monitor/delegates component
- [x] Update monitor/delegates unit test component

### How has this been tested?

1. Run the app.
2. Login to the app using Mainnet or Testnet (Monitor data ONLY works for these networks)
3. Go to Monitor Page
4. Check the Delegates Tab
5. The 3  charts should render properly

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-desktop/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-desktop/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/screens/monitor/delegates/delegates.js [hunk 1]
```diff
@@ -10,22 +10,35 @@ import DelegatesTable from '../../../shared/delegatesTable';
...
+  const [activeTab, setActiveTab] = useState('active');
+  const statuses = {
     forgedThisRound: t('Forging'),
     forgedLastRound: t('Awaiting slot'),
     notForging: t('Not forging'),
     missedLastRound: t('Missed block'),
-  }[status] || t('Loading'));
+  };
+
+  const getForgingTitle = status => statuses[status] || t('Loading');
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/delegates.js]: Why do we need to define this inside? Why does it cover some cases and not all? Why does it return an object from which we later choose a memeber? I think a simple switch-case statement covers this all and returns the value needed directly.

File: src/components/screens/monitor/delegates/delegates.test.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import delegates from '../../../../../test/constants/delegates';
 const transformToLiskServiceFormat = ({ account, ...delegate }) => ({
   ...delegate,
   ...account,
+  status: 'forgedThisRound',
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/delegates.test.js]: If they all have the same status, why bother set it there?

File: src/components/screens/monitor/delegates/delegates.js [hunk 3]
```diff
@@ -37,12 +50,12 @@ const Delegates = ({
       header: t('Address'),
       /* eslint-disable-next-line react/display-name */
       getValue: ({ address }) => <AccountVisualWithAddress {...{ address, isMediumViewPort }} />,
-      className: (filters.tab === 'active'
+      className: (activeTab === 'active'
         ? [grid['col-xs-3'], grid['col-md-3']]
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/delegates.js]: Why we don't use string letrals instead? Cause they're meant for this purpose exactly.

Comment (yasharAyari) [reply src/components/screens/monitor/delegates/delegates.js]: This goes for all other occurances of `Array.join` in this component. My concerns why should we define an array and then merge its members right away?

File: src/components/screens/monitor/delegates/delegates.js [hunk 4]
```diff
@@ -101,19 +114,35 @@ const Delegates = ({
...
+    : !!standByDelegates.data.length && standByDelegates.data.length % DEFAULT_LIMIT === 0;
 
   const getRowLink = delegate => `${routes.accounts.pathPrefix}${routes.accounts.path}/${delegate.address}`;
 
+  const chartDelegatesForging = delegates;
+
+  delegates = activeTab === 'active'
+    ? {
+      ...delegates,
+      data: delegates.data.filter(d => d.username.includes(filters.search || '')),
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/delegates.js]: What do you expect from this filter?
as long as `d.username` is a string, `d.username.includes(filters.search || '')` retruns true no matter what. and if you meant to filter undefined values, this is not the best logic for it.

File: src/components/screens/monitor/delegates/index.js [hunk 5]
```diff
@@ -13,32 +14,70 @@ import withLocalSort from '../../../../utils/withLocalSort';
...
+const standByDelegatesKey = 'standByDelegates';
 
 const mapStateToProps = ({ blocks: { latestBlocks } }) => ({
   latestBlocks,
 });
 
+const transformResponse = (response, oldData, urlSearchParams) => (
+  urlSearchParams.offset
+    ? [...oldData, ...response.filter(
+      delegate => !oldData.find(({ username }) => username === delegate.username),
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/index.js]: IndexOf is alooooooot faster than this iteraton. please read this
https://wsvincent.com/javascript-remove-duplicates-array/

File: src/components/screens/monitor/delegates/index.js [hunk 6]
```diff
@@ -13,32 +14,70 @@ import withLocalSort from '../../../../utils/withLocalSort';
...
+        transformResponse: response => response.total,
+      },
+
+      chartRegisteredDelegatesData: {
+        apiUtil: liskService.getRegisteredDelegates,
+        defaultData: [],
+        autoload: true,
+        transformResponse: (response) => {
+          const responseFormatted = response.reduce((acc, delegate) => {
+            const newDelegate = { ...delegate, timestamp: moment(delegate.timestamp * 1000).startOf('month').toISOString() };
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/index.js]: This part is ambigeous. please explain what are you trying to achive in comments.

File: src/components/screens/monitor/delegates/overview.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+      .reduce((amountOfDelegates, amountOfDelegatesByMonth) => {
+        final.unshift(amountOfDelegates - amountOfDelegatesByMonth);
+        return amountOfDelegates - amountOfDelegatesByMonth;
+      }, totalDelegates);
+
+    return final;
+  };
+
+  const getAmountOfDelegatesLabels = () => {
+    const labels = chartRegisteredDelegates.data.map(date => moment(date.x).format('MMM YY'));
```
Comment (yasharAyari) [inline src/components/screens/monitor/delegates/overview.js]: You just iterated over this array in the paretn component where you retrieved the data, why didn't you format it right away? This iteration is redundant.

**Comment by osvaldovega** [comment]
Thanks for the review Ali, I change the code based on what we discuss. Please review it again and let me know what do you think.

File: src/components/screens/monitor/delegates/delegates.js (hunk 1 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/delegates.js]: we can keep this based on what we discuss

File: src/components/screens/monitor/delegates/delegates.js (hunk 3 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/delegates.js]: fixed, I made the changes

File: src/components/screens/monitor/delegates/delegates.js (hunk 4 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/delegates.js]: we can keep this based on what we discuss

File: src/components/screens/monitor/delegates/index.js (hunk 5 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/index.js]: we can keep this based on what we discuss

File: src/components/screens/monitor/delegates/index.js (hunk 6 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/index.js]: I fixed

File: src/components/screens/monitor/delegates/overview.js (hunk 7 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/overview.js]: I fixed too

File: src/components/screens/monitor/delegates/delegates.test.js (hunk 2 above)
Comment (osvaldovega) [reply src/components/screens/monitor/delegates/delegates.test.js]: we can keep this based on what we discuss

**Pull request merged**